lisa/src/main/java/it/unive/lisa/symbolic/SymbolicExpression.java

```diff
@@ -51,6 +53,33 @@ public final Type getDynamicType() {
 		});
 	}
 
+	/**
+	 * Push a new scope in the call stack caused by calling the method passed as
+	 * parameter. This causes the current local variables to be hidden by the
+	 * method call.
+	 *
+	 * @param scope the called method
+	 * 
+	 * @return the abstract state where the local variables have been hidden
+	 */
+	public abstract SymbolicExpression pushScope(Call scope);
```

Comment:
Is the only purpose of these two method the implementation of `AnalysisState.pushScope()` and `AnalysisState.popScope()`? In this case they have to go away. An expression is just an expression, it does not have a scope stack